gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/pdeps/ResolveProductDependenciesTask.java

```diff
@@ -123,6 +123,9 @@ public final void resolve() throws IOException {
             }
         });
 
+        // Perform initial merge of the discovered dependencies so that if there are overlaps with declared
+        // dependencies we only log a single time.
+        Map<ProductId, ProductDependency> mergedDiscoveredDependencies = new HashMap<>();
```

Comment:
instead of constructing a new map can we merge as we iterate? something like: https://github.com/palantir/sls-packaging/compare/fo/example?expand=1

Reply:
changed as requested, except I took out the "if (mergedDependency.equals(discoveredDependency)))" because there are times when it is not an exact match - e.g. the recommended dependency did not contain a recommended version and that got merged in.